lib/fizzy/parser_expr.cpp

```diff
@@ -619,7 +619,10 @@ parser_result<Code> parse_expr(
         case Instr::i64_store32:
         {
             // alignment
-            std::tie(std::ignore, pos) = leb128u_decode<uint32_t>(pos, end);
+            uint32_t align;
+            std::tie(align, pos) = leb128u_decode<uint32_t>(pos, end);
+            if ((uint64_t{1} << align) > metrics.memory_width)
```

Comment:
You could store `log(max_align)` in the table, then you wouldn't need to shift here.

Otherwise I think you need first also to check `align < 64`, or shift would be undefined behavior.

Reply:
You know what I'll just go back to the original commit 975fe8a677.

```
            // NOTE: align > 4 is the hard limit (64 / 8) >> 1, and checking it to avoid overflows
            if ((align > 4) || ((1 << align) > metrics.memory_width))
```

Reply:
Added back the correct version (align <= 3).